jupyterlab_gcedetails/src/components/auth_token_retrieval.tsx

```diff
@@ -1,21 +1,30 @@
-export function AuthTokenRetrieval(callback: (err, credentials) => void) {
+export function authTokenRetrieval(): Promise<string> {
   // eslint-disable-next-line prefer-const
   let authPopup: any;
   const authOrigin = 'https://jupyterlab-interns-sandbox.uc.r.appspot.com';
-  const listener = (message: MessageEvent) => {
-    if (
-      message.origin === authOrigin &&
-      (message.data['error'] || message.data['credentials'])
-    ) {
-      authPopup.close();
-      window.removeEventListener('message', listener);
-      callback(message.data['error'], message.data['credentials']);
+  return new Promise<string>((resolve, reject) => {
+    try {
+      const listener = (message: MessageEvent) => {
+        if (
+          message.origin === authOrigin &&
+          (message.data['error'] || message.data['credentials'])
+        ) {
+          authPopup.close();
+          window.removeEventListener('message', listener);
+          if (message.data['error']) {
+            throw new Error();
```

Comment:
To simplify and eliminate the need for the try/catch, I think you could also just call `reject(message.data['error'[)` and then return. I would suggest adding the error information to the rejection so you could potentially surface what failed.